proposal/color-4-new-spaces-js.d.ts.md

```diff
@@ -0,0 +1,540 @@
+```ts
+isChannelPowerless(options: {
+  channel: ChannelName;
+  space?: KnownColorSpace;
+}): boolean;
+```
+
+#### `toSpace`
+
+Returns the result of converting [`internal`] to `space` as a new SassColor.
+
+```ts
+toSpace(space: KnownColorSpace): SassColor;
+```
+
+#### `toGamut`
+
+Returns [`internal`] constrained to its space's gamut as a new SassColor. This
+is generally not recommended since even older browsers will display out-of-gamut
+colors as best they can, but it may be necessary in some cases.
+
+```ts
+toGamut(): SassColor;
+```
+
+#### `changeChannels`
+
+Returns a new SassColor as the result of changing some of [`internal`]'s
+channels. The `space` value defaults to the `space` of [`internal`], and any
+combination of channels in that space may be changed. Throws an error if any
+`channel` is not present in `space`.
+
+* If `space` is not defined, let `space` be the value of [`space(internal)`].
+
+* Let `arguments` be the key value pairs in `channels` as keyword arguments.
+
+* Return the value of the result of [`change(internal, ...arguments)`].
+
+[`space(internal)`]: #space
+
+[`change(internal, ...arguments)`]: ./color-4-new-spaces.md#colorchange
+
+```ts
+changeChannels(
+  channels: {
+    [key in ChannelNameRGB]?: ChannelValue;
+  },
+  space?: ColorSpaceRGB
+): SassColor;
+
+changeChannels(
+  channels: {
+    [key in ChannelNameHWB]?: ChannelValue;
+  },
+  space?: ColorSpaceHWB
+): SassColor;
+
+changeChannels(
+  channels: {
+    [key in ChannelNameHSL]?: ChannelValue;
+  },
+  space?: ColorSpaceHSL
+): SassColor;
+
+changeChannels(
+  channels: {
+    [key in ChannelNameXYZ]?: ChannelValue;
+  },
+  space?: ColorSpaceXYZ
+): SassColor;
+
+changeChannels(
+  channels: {
+    [key in ChannelNameLCH]?: ChannelValue;
+  },
+  space?: ColorSpaceLCH
+): SassColor;
+
+changeChannels(
+  channels: {
+    [key in ChannelNameLAB]?: ChannelValue;
+  },
+  space?: ColorSpaceLAB
+): SassColor;
+```
+
+### `interpolate()`
+
+Returns a new SassColor with the result of mixing [`internal`] with `color2`.
+
+It accepts an optional float `weight`, which defaults to 0.5. Lower values will
+appear closer to [`internal`] and higher values will appear closer to `color2`.
+
+If `space` is set, interpolation will happen in that space. Otherwise it will
+happen in the color space for [`internal`].
+
+If `space` (or the color space of [`internal`] if no `space` argument is
+provided) is a PolarColorSpace (a color space with a polar angle `hue` channel),
+a `method` may be provided, which defaults to `shorter`.
+
+```ts
+interpolate(options: {
+  color2: SassColor;
+  weight?: number;
+  space?: RectangularColorSpace;
+}): SassColor;
+
+interpolate(options: {
+  color2: SassColor;
+  weight?: number;
+  space?: PolarColorSpace;
+  method?: HueInterpolationMethod;
+}): SassColor;
+```
+
+[`internal`]: ../spec/js-api/value/color.d.ts.md#internal
+
+### New Constructors
+
+Because the value of each channel may be a string, number, or `null`, this
```

Comment:
If you're going to define this in terms of Sass color functions, you'll need to have some logic to explicitly convert the JS `null` to the unquoted Sass string `none`.